application/src/main/java/org/togetherjava/tjbot/commands/moderation/scam/ScamDetector.java

```diff
@@ -0,0 +1,105 @@
+package org.togetherjava.tjbot.commands.moderation.scam;
+
+import org.jetbrains.annotations.NotNull;
+import org.togetherjava.tjbot.commands.utils.StringDistances;
+import org.togetherjava.tjbot.config.Config;
+import org.togetherjava.tjbot.config.ScamBlockerConfig;
+
+import java.net.URI;
+import java.util.regex.Pattern;
+
+public class ScamDetector {
+    private static final Pattern TOKENIZER = Pattern.compile("[\\s,]");
+    private final ScamBlockerConfig config;
+
+    public ScamDetector(@NotNull Config config) {
+        this.config = config.getScamBlocker();
+    }
+
+    public boolean isScam(@NotNull CharSequence message) {
+        AnalyseResults results = new AnalyseResults();
+        TOKENIZER.splitAsStream(message).forEach(token -> analyzeToken(token, results));
+        return isScam(results);
+    }
+
+    private boolean isScam(@NotNull AnalyseResults results) {
+        if (results.pingsEveryone && results.containsNitroKeyword && results.hasUrl) {
+            return true;
+        }
+        return results.containsNitroKeyword && results.hasSuspiciousUrl;
+    }
+
+    private void analyzeToken(@NotNull String token, @NotNull AnalyseResults results) {
+        if ("@everyone".equalsIgnoreCase(token)) {
+            results.pingsEveryone = true;
+        }
+        if ("nitro".equalsIgnoreCase(token)) {
```

Comment:
```suggestion
        }
        
        if ("nitro".equalsIgnoreCase(token)) {
```
Make sure that the if statements are vaccinated, preferably even with a booster.
Otherwise keep the distance